### PR TITLE
The first step in replacing newtonsoft with System.Text.Json

### PR DESCRIPTION
**Pull request by brentschmaltz**
Removed our internal newtonsoft, all our code uses the public newtonsoft.
This will ease the work to use System.Json.Text on a class-by-class basis.
The use of newtonsoft will be removed in all src code when we ship, but may remain in test code.

Removed tests that were designed to ensure that changes from 4.x -> 5.x where switched from System.IdentityModel.Tokens to Microsoft.IdentityModel.Tokens were not breaking, we do not need this code anymore.

Removed some old performance code that was moved to its own repo.
Cleaned up 'usings'.
Set CLSCompliant to false in all test code (Assemblyinfo.cs) so #ifdef's could be removed in code.

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: build/common.props [hunk 1]
```diff
@@ -45,6 +45,7 @@
   
   <ItemGroup Condition="'$(TargetFramework)' != 'net8.0'">
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="$(MicrosoftSourceLinkGitHubVersion)" PrivateAssets="All"/>
+    <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftVersion)" />
```
Comment (eerhardt) [inline build/common.props:52]: Does this code build if you are targeting `net8.0`? How does this PackageReference come in that case?

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 using System.Runtime.InteropServices;
 using Xunit;
 
-[assembly: CLSCompliant(true)]
+[assembly: CLSCompliant(false)]
```
Comment (eerhardt) [inline test/Microsoft.IdentityModel.JsonWebTokens.Tests/Properties/AssemblyInfo.cs:8]: Why are these CLSCompliant attributes changing?

Comment (brentschmaltz) [reply test/Microsoft.IdentityModel.JsonWebTokens.Tests/Properties/AssemblyInfo.cs:8]: We don't need to be CLSCompliant in test.

File: build/common.props (hunk 1 above)
Comment (brentschmaltz) [reply build/common.props:52]: I will have to fix that.

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (eerhardt) [reply test/Microsoft.IdentityModel.JsonWebTokens.Tests/Properties/AssemblyInfo.cs:8]: I would just remove these lines then. If you don't have a CLSCompliant attribute, the assembly is not CLSCompliant.

**Review by westin-m — APPROVED** [review]

**Pull request merged**